File: leetcode/by_companies/facebook/lp1480.py
class Solution:
    def runningSum(self, nums: list[int]) -> list[int]:
        rs = [0] * len(nums)

        for i, num in enumerate(nums):
            if i > 0:
                rs[i] += rs[i - 1]

            rs[i] += num

        return rs

File: javascript/src/leetcode/by-topic/prefix-sum/lp1480.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const runningSum = function (nums) {
  const result = []

  for (let i = 0; i < nums.length; i++) {
    if (i === 0) {
      result.push(nums[i])
    } else {
      result.push(result.at(-1) + nums[i])
    }
  }

  return result
}

module.exports = { runningSum }
